pkg/controllers/provisioning/provisioner.go

```diff
@@ -215,6 +217,11 @@ func (p *Provisioner) NewScheduler(ctx context.Context, pods []*v1.Pod, stateNod
 	instanceTypes := map[string][]*cloudprovider.InstanceType{}
 	domains := map[string]sets.Set[string]{}
 	for _, nodePool := range nodePoolList.Items {
+		nodePoolReady := nodePool.StatusConditions().Get(status.ConditionReady)
+		if !nodePoolReady.IsTrue() {
+			logging.FromContext(ctx).With("nodepool", nodePool.Name).Errorf("skipping, nodepool not ready")
```

Comment:
Is this an error or a debug log? I think a debug log, personally. Since this is already surfaced through the readiness of the NodePool, I think we can just add this as additional info, but not something that we should be screaming about in our logs